Classes/Figure.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Classes
+{
+    //3
+    internal class Figure
+    {
+        private string nameFigure;
+        private double perimeter;
+        private Point[] point = new Point[3];
+
+        public Figure(string nameFigure, Point point1, Point point2, Point point3)
+        {
+            this.nameFigure = nameFigure;
+            point[0] = point1;
+            point[1] = point2;
+            point[2] = point3;
+        }
+
+        public Figure(string nameFigure, Point point1, Point point2, Point point3, Point point4)
+        {
+            this.nameFigure = nameFigure;
+            ResizePlusOnePointArray(ref point);
+            point[0] = point1;
+            point[1] = point2;
+            point[2] = point3;
+            point[3] = point4;
+        }
+
+        public Figure(string nameFigure, Point point1, Point point2, Point point3, Point point4, Point point5)
+        {
+            this.nameFigure = nameFigure;
+            ResizePlusOnePointArray(ref point);
+            ResizePlusOnePointArray(ref point);
+            point[0] = point1;
+            point[1] = point2;
+            point[2] = point3;
+            point[3] = point4;
+            point[4] = point5;
+        }
+
+        public double CalculateLengthSide(Point X, Point Y)
+        {
+            double Length = Math.Sqrt(Math.Pow((Convert.ToDouble(Y.xPoint - X.xPoint)), 2)
+                                    + Math.Pow((Convert.ToDouble(Y.yPoint - X.yPoint)), 2));
+            return Length;
+        }
+
+        private void ResizePlusOnePointArray(ref Point[] arr)
+        {
+            Array.Resize(ref arr, arr.Length + 1);
+        }
+
+        public void CalculatePerimeter()
```

Comment:
Rewrite this method and use loops. Don't check for `nameFigure` just do `foreach` instead